### ISSUE TITLE
component name clashes

### ISSUE THREAD

**Issue by valgarf** [zauberzeug/nicegui#1887]
### Description

If two components only differ in the last 4 letters, the generated components' template id is identical. For example, if you have two components named `kanban_list.vue` and `kanban_card.vue`, the generated javascript code is:
```
var kanban_card = app.component('nicegui-kanban_card', {template:'#tpl-kanban_', ...
var kanban_list = app.component('nicegui-kanban_list', {template:'#tpl-kanban_', ...
```
One of these components is using the wrong template resulting in some strange errors.

The reason is the `register_vue_component` function in dependencies.py. It calls `_get_name`, which strips off everything after the first dot. This name is then given to `vbuild.VBuild`, which expects a filename ending in ".vue" and indiscriminately removes the last 4 letters.

**Comment by falkoschindler** [comment]
Oh wow, what a nasty bug! Thanks for reporting this issue, @valgarf!

So it looks like
```py
v = vbuild.VBuild(path.name, path.read_text())
```
fixes it.

Do you want to like to create a pull request, becoming an official contributor? 🙂

**Comment by valgarf** [comment]
If it can wait until tomorrow, I will gladly create a pull request then.

**Comment by falkoschindler** [comment]
@valgarf Sorry for being a bit impatient, but I quickly fixed it myself since we would like to have it being part a the urgent bugfix release 1.4.1. 🚀 It will fix a major bug we introduced with 1.4.0.
